diff --git a/node_modules/ts-poet/build/Import.js b/node_modules/ts-poet/build/Import.js
index b92bea3..0464ac9 100644
--- a/node_modules/ts-poet/build/Import.js
+++ b/node_modules/ts-poet/build/Import.js
@@ -26,6 +26,6 @@ exports.sameModule = exports.maybeRelativePath = exports.emitImports = exports.S
 const lodash_1 = __importDefault(require("lodash"));
 const path = __importStar(require("path"));
 const Node_1 = require("./Node");
 const typeImportMarker = '(?:t:)?';
 const fileNamePattern = '(?:[a-zA-Z0-9._-]+)';
 const modulePattern = `@?(?:(?:${fileNamePattern}(?:/${fileNamePattern})*))`;
@@ -293,6 +294,18 @@ function emitImports(imports, ourModulePath, importMappings) {
         return '';
     }
     let result = '';
+    // HACK: take the project root import path from $PROJECT
+    const thisProject = process.env.PROJECT;
+    let ourModuleImportPath = path.normalize(ourModulePath);
+    // HACK: if this is an import from our project, set the path accordingly
+    // github.com/aperturerobotics/protobuf-project/example/example -> ./example/example
+    if (thisProject) {
+        if (ourModuleImportPath.startsWith(thisProject)) {
+            ourModuleImportPath = './' + ourModuleImportPath.substr(thisProject.length + 1);
+        } else {
+            ourModuleImportPath = './vendor/' + ourModuleImportPath;
+        }
+    }
     const augmentImports = lodash_1.default.groupBy(filterInstances(imports, Augmented), (a) => a.augmented);
     // Group the imports by source module they're imported from
     const importsByModule = lodash_1.default.groupBy(imports.filter((it) => it.source !== undefined &&
@@ -307,7 +320,18 @@ function emitImports(imports, ourModulePath, importMappings) {
         if (modulePath in importMappings) {
             modulePath = importMappings[modulePath];
         }
-        const importPath = maybeRelativePath(ourModulePath, modulePath);
+        if (thisProject) {
+            if (modulePath.startsWith('./')) {
+                if (modulePath.substring(2).startsWith(thisProject)) {
+                    modulePath = './' + modulePath.substring(3 + thisProject.length);
+                } else if (modulePath.startsWith('./google/protobuf')) {
+                    modulePath = '@aperturerobotics/ts-proto-common-types/' + path.normalize(modulePath);
+                } else {
+                    modulePath = './vendor/' + path.normalize(modulePath);
+                }
+            }
+        }
+        const importPath = maybeRelativePath(ourModuleImportPath, modulePath);
         // Output star imports individually
         unique(filterInstances(imports, ImportsAll).map((i) => i.symbol)).forEach((symbol) => {
             result += `import * as ${symbol} from '${importPath}';\n`;
